packages/mix-models/src/rollback.ts

```diff
@@ -4,9 +4,10 @@ import set from 'lodash/set';
 
 import { Constructor } from './model';
 import { flattenKeys } from './flatten-keys';
+import { isArray } from 'lodash';
 
 export type RollbackMask = {
```

Comment:
This type does not represents a real rollback mask type. Look at the `Pattern` interface of validation mixin.

Reply:
My suggestion is:

```ts
export interface RollbackArrayMask extends Record<string, RollbackMask | boolean | number | number[] | undefined> {
  $array: true;
  $index?: number | number[];
}

export interface RollbackMask {
  [key: string]: RollbackMask | RollbackArrayMask | boolean;
}
```

Reply:
Maybe,

```ts
export type RollbackArrayMask = RollbackMask & { $array: true; $index?: number[] };

export type RollbackMask = { [key: string]: RollbackMask | RollbackArrayMask | boolean };
```
?